components/BottomMenu.js

```diff
@@ -38,61 +41,93 @@ function BottomMenu ({ navigation }, props) {
+    }
 
     useEffect(() => {
         const intervalId = setInterval(() => {
             getBuildingSelected();
             getDestination();
+            getPersonaType();
+            getMobility();
+            getMethodTravel();
             //getMapPressed();
         }, 1);
         return () => clearInterval(intervalId);
     });
 
-    if (iconSelected && selectedBuilding) {
-        return (
-            <View style={styles.moreDetails}>
-                <MoreDetails name={selectedBuilding} navigation={navigation} />
-                <Icon name="ios-arrow-down" style={styles.arrowDown} onPress={() => { setIconSelected(false); }} />
-            </View>
-        );
-    }
-    if (iconSelected && !selectedBuilding) {
-        return (
-            <View style={styles.moreDetails}>
-                <Icon name="ios-arrow-down" style={styles.arrowDown} onPress={() => { setIconSelected(false); }} />
-            </View>
-        );
-    }
+    const nameMethodTravel = () => {
+        switch (methodTravel) {
+            case 'driving':
+                return "Driving";
 
-    const goToDoubleSearchBar = () => {
-        navigation.navigate("DoubleSearch", { destinationName: destination });
-    };
+            case 'walking':
+                return "Walking";
+
+            case 'transit':
+                return "Transit";
 
-    if (getInside) {
+            case 'bicycling':
+                return "Bicycling";
+
+            default:
+                return "Driving";
+        }
+        // return methodTravel;
+    }
+
+    if (previewDirections) {
         return (
-            <View style={styles.insideBuildingContainer}>
-                <Icon name="ios-arrow-up" style={styles.arrowUp} onPress={() => { setIconSelected(true); }} />
-                <Text style={styles.mainLabel}>{selectedBuilding}</Text>
-                <Text style={styles.shortLabel}>More info</Text>
-                <View style={styles.btnleave}>
-                    <Button style={styles.btnleave} color={"#3ACCE1"} uppercase={false} mode="contained" onPress={() => {
-                        setGetInside(false);
-                    }}>
-                        <Text style={styles.btnText}>Exit Building</Text>
-                    </Button>
-                </View>
-                <View style={styles.changeFloor}>
-                    <FloorMenu />
+            <View style={styles.container}>
+                <Icon name="ios-arrow-up" style={styles.arrowUp} onPress={goToPreferenceMenu} />
+                <Text style={styles.mainLabel}>{props.directionResponse ? props.directionResponse.generalRouteInfo.totalDuration : "N/A"} ({props.directionResponse ? props.directionResponse.generalRouteInfo.totalDistance : "N/A"})</Text>
+                <Text style={styles.shortLabel}>Main Travel Mode: {nameMethodTravel()}</Text>
+                <View style={styles.btnGetDirection}>
+                <Button style={{ left: "100%", }}
```

Comment:
avoid inline styling please

Reply:
Adding another style category in the **stylesheet** for one line will make the styling sections will large though

Reply:
I'm going to add it to the styling section for consistency anyway